libp2p/security/insecure/transport.py

```diff
@@ -24,12 +64,30 @@ class InsecureTransport(BaseSecureTransport):
         for an inbound connection (i.e. we are not the initiator)
         :return: secure connection object (that implements secure_conn_interface)
         """
-        return InsecureSession(self, conn, ID(b""))
+        session = InsecureSession(self, conn, ID(b""))
+        await session.run_handshake()
+        return session
 
     async def secure_outbound(self, conn: IRawConnection, peer_id: ID) -> ISecureConn:
         """
         Secure the connection, either locally or by communicating with opposing node via conn,
         for an inbound connection (i.e. we are the initiator)
         :return: secure connection object (that implements secure_conn_interface)
         """
-        return InsecureSession(self, conn, peer_id)
+        session = InsecureSession(self, conn, peer_id)
+        await session.run_handshake()
+        received_peer_id = session.get_remote_peer()
+        if session.remote_permanent_pubkey is not None and received_peer_id != peer_id:
```

Comment:
IMO this check should be handled in `run_handshake`(L50).

Reply:
The reason why I do it in `secure_outbound` is, we only need to check "whether the `PeerID` of node we are dialing is the same as the one it send to us" in `secure_outbound`. If we put it in `run_handshake`, we will need to add redundant checks like "whether we are the initiator or not".

Reply:
On second thought, I move all these logic inside `run_handshake`. Thanks for the idea.